File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-3.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int len = 0;
        ListNode lenC = head;
        while(lenC != null){
            len++;
            lenC = lenC.next;
        }
        if(len-n == 0){
            head = head.next;
            return head;
        }
        ListNode curr = head;
        ListNode prev = null;
        ListNode end= null;
        int currV = 0;
        while(currV != len-n){
            currV++;
            ListNode nC = curr.next;
            prev = curr;
            curr = nC;
            nC = nC.next;
            end = nC;
        }
        if(prev == null){
            curr = curr.next;;
        }
        if(end == null){
            prev.next = end;
        }else{
            curr.next = null;
            prev.next = end;
        }
        return head;
    }
}

File: Data Structures & Algorithms/reverse-a-linked-list/submission-3.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return None 
            
        prev = None
        curr = head
        currN = curr.next

        while currN:
            curr.next = prev
            prev = curr
            curr = currN
            currN = currN.next
        
        curr.next = prev
        
        return curr